### PR TITLE
Simplify wrappers

### PR DESCRIPTION
**Pull request by ParnianAlimi**
Part of CRYP-207.

### CODE REVIEW

**Review by fmiguelgarcia — APPROVED** [review]

File: cryptography-core/src/asset_proofs/macros.rs [hunk 1]
```diff
@@ -74,3 +74,21 @@ macro_rules! define_sub_assign_variants {
         }
     };
 }
+
+/// That `ensure` does not transform into a string representation like `failure::ensure` is doing.
```
Comment (aafshar-poly) [inline cryptography-core/src/asset_proofs/macros.rs:78]: Which `ensure`?

File: cryptography-core/src/asset_proofs/membership_proof.rs [hunk 2]
```diff
@@ -4,6 +4,7 @@
 //! This implementation is based on one-out-of-many proof construction described in the following paper
 //! <https://eprint.iacr.org/2015/643.pdf>
 
+use super::errors::{ErrorKind, Fallible};
```
Comment (aafshar-poly) [inline cryptography-core/src/asset_proofs/membership_proof.rs:7]: Why `super` vs `crate`? This is a general question not specifically for this line.

File: cryptography-core/src/dalek_wrapper/mod.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
+use super::cdd_claim::{RISTRETTO_POINT_SIZE, SCALAR_SIZE}; // todo: maybe move these here.
+use codec::{Decode, Encode, Error as CodecError, Input, Output};
+use curve25519_dalek::{
+    ristretto::{CompressedRistretto, RistrettoPoint},
+    scalar::Scalar,
```
Comment (aafshar-poly) [inline cryptography-core/src/dalek_wrapper/mod.rs:5]: change these to `scalar::Scalar as DalekScalar` and then, you can give the new types Scalar name. E.g., `struct Scalar(pub DalekScalar)`. Similarly for RistrettoPoint

**Review by aafshar-poly — COMMENTED** [review]
Nice work, added a comment on reusing the Scalar and RisttrettoPoint names.

File: cryptography-core/src/asset_proofs/membership_proof.rs (hunk 2 above)
Comment (ParnianAlimi) [reply cryptography-core/src/asset_proofs/membership_proof.rs:7]: `crate` refers to the root scope vs `super` refers to the parent scope. here we could do `crate::asset_proofs::errors::{ErrorKind, Fallible};` instead.

File: cryptography-core/src/asset_proofs/macros.rs (hunk 1 above)
Comment (ParnianAlimi) [reply cryptography-core/src/asset_proofs/macros.rs:78]: very good question :-? I moved this here from a lib.rs don't have any idea what it is referring to.

**Review by aafshar-poly — APPROVED** [review]
lgtm :+1:

**Pull request merged**